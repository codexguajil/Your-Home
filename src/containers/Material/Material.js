import React, { Component } from 'react';
import EditMaterial from '../EditMaterial/EditMaterial';
import DeleteConfirm from '../../components/DeleteConfirm/DeleteConfirm';
import MaterialDetails from '../../components/MaterialDetails/MaterialDetails';

class Material extends Component {

  render() {
    const { brand, name } = this.props.material
    return (
      <div key={name} className='material'>
        <p>{name}, {brand}</p>
        <div className='material-control'>
          <MaterialDetails {...this.props.material}/>
          <EditMaterial {...this.props.material}/>
          <DeleteConfirm type="material"/>
        </div>
      </div>
    );
  }
}

export default Material;