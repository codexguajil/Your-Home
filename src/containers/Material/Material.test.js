import React from 'react';
import { shallow } from 'enzyme';
import Material from './Material';

describe('Material', () => {
  let wrapper;
  let mockMaterial;
  beforeEach(() => {
    mockMaterial = {
      "id": 1,
      "element_type": "Flooring",
      "material": {
          "id": 1,
          "name": "Material 1",
          "model_number": "abc1",
          "brand": "Kenmoore",
          "vendor": "HD",
          "manual_url": null,
          "notes": null,
          "quantity": null,
          "unit_price": null
      }
  }
    wrapper = shallow(<Material material={mockMaterial} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});