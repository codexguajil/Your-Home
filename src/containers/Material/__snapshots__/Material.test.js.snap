// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material should match snapshot 1`] = `
<div
  className="material"
>
  <p>
    , 
  </p>
  <div
    className="material-control"
  >
    <MaterialDetails
      element_type="Flooring"
      id={1}
      material={
        Object {
          "brand": "Kenmoore",
          "id": 1,
          "manual_url": null,
          "model_number": "abc1",
          "name": "Material 1",
          "notes": null,
          "quantity": null,
          "unit_price": null,
          "vendor": "HD",
        }
      }
    />
    <EditMaterial />
    <DeleteConfirm
      type="material"
    />
  </div>
</div>
`;
